specs/EmergencyAuthCodeAPI.yml

```diff
@@ -0,0 +1,226 @@
+  /emergency-auth-code-service/company/{company_number}/officers/{officer_id}:
+    parameters:
+      - $ref: '#/components/parameters/companyNumber'
+      - $ref: '#/components/parameters/officerId'
+    get:
+      tags:
+        - officers
+      operationId: getCompanyOfficer
+      summary: Get a specific company officer that is eligible for emergency auth code delivery
+      responses:
+        '200':
+          description: An eligible officer
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/companyOfficer'
+        '401':
+          description: Unauthorised
+        '404':
+          description: Not found
+  /emergency-auth-code-service/auth-code-requests:
+    post:
+      tags:
+        - auth-code-requests
+      operationId: createAuthCodeRequest
+      summary: Create an emergency auth code request
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        description: Emergency auth code request data
+        required: true
+      responses:
+        '201':
+          description: Created emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+  /emergency-auth-code-service/auth-code-requests/{auth_code_request_id}:
+    parameters:
+      - $ref: '#/components/parameters/authCodeRequestId'
+    get:
+      tags:
+        - auth-code-requests
+      operationId: getAuthCodeRequest
+      summary: Get an emergency auth code request
+      responses:
+        '200':
+          description: Updated emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+    put:
+      tags:
+        - auth-code-requests
+      operationId: updateAuthCodeRequest
+      summary: Update an emergency auth code request
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        description: Emergency auth code request data
+        required: true
+      responses:
+        '200':
+          description: Updated emergency auth code request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/emergencyAuthCodeRequest'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorised
+components:
+  schemas:
+    companyOfficer:
+      type: object
+      required:
+        - id
+        - forename
```

Comment:
I've just been playing around with returning CHIPS information and it seems like the officers have a forename1 and forename2 on the officer_details table. Should both be captured here as well?

Reply:
Possibly, I've gone with very basic placeholder data models at this point as I haven't had time to get into the required data model requirements in oracle and we don't have much in terms of UX at the moment so holding back on adding too much detail that we may need to either rework or remove later on e.g. service addresses, dob etc have been left out for now until we know if they will be required for the UX side of the service. Other parts have purposely been left out for now like the actual URA that we won't expose via the API but will be used behind the scenes.

Reply:
After seeing the prototype requirements I've added extra data to the officer models and simplified the "officer name" to a single field after cross referencing against our output data specs which combine all name elements into a single field already - these specs do sometimes offer a breakdown of name elements such as title, forename, other forename, surname, honours, etc. which can be added here if required later on but given the prototype screens as they are I don't see the need for this breakdown at this point